core/src/main/java/org/infinispan/util/ByteString.java

```diff
@@ -39,7 +38,7 @@ public static ByteString emptyString() {
 
    @Override
    public int hashCode() {
-      return hash;
+      return hash == 0 ? (hash = Arrays.hashCode(b)) : hash;
```

Comment:
no. the first thing we do when we create it, is to put it in a map or check it in a map :P